### PR TITLE
Prep for release after Eve Online merge

### PR DESCRIPTION
**Pull request by pbrisbin**
Fixes to EveOnline plugin

- Fix invalid data declaration
- Fix -Wall compilation warnings
  - Unused imports
  - Unused functions
  - Unused arguments
  - Shadowed function name
- Incorporate plain link into WidgetType, remove Maybe
- Use only one function, always accept WidgetType

Add/fix documentation, change argument order

- Document the difference between widget and non-widget functions
- Pass the widget as the first argument to take advantage of partial
  application

/cc @Drezil

### CODE REVIEW

File: Yesod/Auth/OAuth2/EveOnline.hs [hunk 1]
```diff
@@ -35,18 +33,19 @@ import Yesod.Core.Widget
...
+    = Plain -- ^ Simple "Login via eveonline" text
+    | BigWhite
     | SmallWhite
     | BigBlack
     | SmallBlack
     | Custom (WidgetT m IO ())
 
 data EveUser = EveUser
-    { eveUserId :: Int
+    { _eveUserId :: Int
```
Comment (jferris) [inline Yesod/Auth/OAuth2/EveOnline.hs]: Was this shadowing something? If the field is unused, can we drop it entirely?

**Comment by pbrisbin** [comment]
The were being parsed out of the JSON but otherwise unused (hence the warning). I've removed them entirely now.

**Comment by pbrisbin** [comment]
Ping @jferris any reason not to merge this?

**Comment by jferris** [comment]
Nope - looks good to merge.

**Pull request merged**